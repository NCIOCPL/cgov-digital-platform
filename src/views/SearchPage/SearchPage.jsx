import React, { useEffect, useRef } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Delighter, StickySubmitBlock } from '../../components/atomic';
import {
  Age,
  CancerTypeCondition,
  CancerTypeKeyword,
  DrugTreatment,
  KeywordsPhrases,
  LeadOrganization,
  Location,
  TrialId,
  TrialInvestigators,
  TrialPhase,
  TrialType,
  ZipCode,
} from '../../components/search-modules';
import { updateForm } from '../../store/actions';

//Module groups in arrays will be placed side-by-side in the form
const basicFormModules = [CancerTypeKeyword, [Age, ZipCode]];
const advancedFormModules = [
  CancerTypeCondition,
  [Age, KeywordsPhrases],
  Location,
  TrialType,
  DrugTreatment,
  TrialPhase,
  TrialId,
  TrialInvestigators,
  LeadOrganization,
];

const SearchPage = form => {
  const dispatch = useDispatch();
  const sentinelRef = useRef(null);
  const formType = useSelector(store => store.form.formType);

  // scroll to top on mount
  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);

  const handleUpdate = (field, value) => {
    dispatch(
      updateForm({
        field,
        value,
      })
    );
  };

  const renderDelighters = () => (
    <div className="cts-delighter-container">
      <Delighter
        classes="cts-livehelp"
        url="/contact"
        titleText={
          <>
            Have a question?
            <br />
            We're here to help
          </>
        }
      >
        <p>
          <strong>Chat with us:</strong> LiveHelp
          <br />
          <strong>Call us:</strong> 1-800-4-CANCER
          <br />
          (1-800-422-6237)
        </p>
      </Delighter>

      <Delighter
        classes="cts-what"
        url="/about-cancer/treatment/clinical-trials/what-are-trials"
        titleText={<>What Are Cancer Clinical Trials?</>}
      >
        <p>Learn what they are and what you should know about them.</p>
      </Delighter>

      <Delighter
        classes="cts-which"
        url="/about-cancer/treatment/clinical-trials/search/trial-guide"
        titleText={<>Which trials are right for you?</>}
      >
        <p>
          Use the checklist in our guide to gather the information you’ll need.
        </p>
      </Delighter>
    </div>
  );

  const toggleForm = () => {
    handleUpdate('formType', formType === 'basic' ? 'advanced' : 'basic');
  };

  const renderSearchTip = () => (
    <div className="cts-search-tip">
      <div className="cts-search-tip__icon">
        <i className="lightbulb-circle-yellow"></i>
      </div>
      <div className="cts-search-tip__body">
        <strong>Search Tip:</strong>
        {formType === 'basic' ? (
          <>{` For more search options, use our `}</>
        ) : (
          <>{` All fields are optional. Skip any items that are unknown or not applicable or try our `}</>
        )}
        <button type="button" className="btnAsLink" onClick={toggleForm}>
          {formType === 'basic' ? 'advanced search' : 'basic search'}
        </button>
        .
      </div>
    </div>
  );

  let formModules =
    formType === 'advanced' ? advancedFormModules : basicFormModules;

  const handleSubmit = e => {
    e.preventDefault();
  };

  return (
    <article className="search-page">
      <div ref={sentinelRef} className="search-page__sentinel"></div>
      <div className="search-page__header">
        <p>
          NCI-supported clinical trials are those sponsored or otherwise
          financially supported by NCI. See our guide, Steps to Find a Clinical
          Trial, to learn about options for finding trials not included in NCI's
          collection.
        </p>
        {renderSearchTip()}
      </div>

      <div className="search-page__content">
        <form
          onSubmit={handleSubmit}
          className={`search-page__form ${formType}`}
        >
          {formModules.map((Module, idx) => {
            if (Array.isArray(Module)) {
              return (
                <div key={`formAdvanced-${idx}`} className="side-by-side">
                  {Module.map((Mod, i) => (
                    <Mod
                      key={`formAdvanced-${idx}-${i}`}
                      handleUpdate={handleUpdate}
                    />
                  ))}
                </div>
              );
            } else {
              return (
                <Module
                  key={`formAdvanced-${idx}`}
                  handleUpdate={handleUpdate}
                />
              );
            }
          })}

          <StickySubmitBlock sentinel={sentinelRef} />
        </form>
        <aside className="search-page__aside">{renderDelighters()}</aside>
      </div>

      <div className="search-page__footer">
        <div className="api-reference-section">
          <h3 id="ui-id-4">
            The Clinical Trials API: Use our data to power your own clinical
            trial search
          </h3>
          <p className="api-reference-content">
            An application programming interface (API) helps translate large
            amounts of data in meaningful ways. NCI’s clinical trials search
            data is now powered by an API, allowing programmers to build
            applications <a href="/syndication/api">using this open data.</a>
          </p>
        </div>
      </div>
    </article>
  );
};

export default SearchPage;
