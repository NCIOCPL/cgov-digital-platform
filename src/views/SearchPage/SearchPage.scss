.search-page {
  &__header {
    margin-bottom: 30px;
  }

  &__content {
    @media screen and (min-width: 1025px) {
      display: flex;
      flex-wrap: nowrap;
      justify-content: space-between;
    }
  }
  &__form {
    margin: 0 1% 30px 0;

    @media screen and (min-width: 1025px) {
      flex: 1 1 75%;
    }
  }
  &__aside {
    margin-bottom: 30px;
    @media screen and (min-width: 1025px) {
      flex: 0 0 24%;
      margin-left: 1%;
    }
  }
  &__footer {
    // the following will be available when it finally resides within the site
    .api-reference-section {
      @media screen and (min-width: 1025px) {
        width: 75%;
      }
      h3 {
        background-color: #319ebd;
        color: #fff;
        margin: 0;
        padding: 15px;
        border-bottom: 10px solid #167a94;
      }
      .api-reference-content {
        position: relative;
        padding: 25px 25px 25px 105px;
        background-color: #f3f3f3;
        min-height: 100px;
        margin: 0;
        font-size: 14px;

        &::before {
          content: '';
          position: absolute;
          width: 4.1875em;
          height: 4.18756em;
          background-position: 8.06452% 1.33107%;
          left: 2em;
          top: 23px;
          display: inline-block;
          background-image: url($imgPath+'/publishedcontent/images/images/design-elements/sprites/svg-sprite.__v15577802.svg');
          background-size: 12em 262.5em;
        }
      }
    }
  }
}

.cts-search-tip {
  display: flex;
  flex-wrap: nowrap;
  align-items: center;
  border: 1px solid #bdbdc2;
  padding: 16px;
  margin-bottom: 1em;

  &__icon {
    width: 3.9375em;
    height: 3.9375em;
    background-position: 7.8125% 52.90135%;
    min-width: 3.9375em;
    display: inline-block;
    background-image: url($imgPath+'/publishedcontent/images/images/design-elements/sprites/svg-sprite.__v15577802.svg');
    background-size: 12em 262.5em;
  }
  &__body {
    padding-left: 1em;

    button {
      appearance: none;
      border: 0;
      padding: 0;
      font-size: 1em;
      color: #2b7bba;
      &:focus,
      &:hover {
        text-decoration: underline;
      }
    }
  }
}

.cts-delighter-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 100%;

  @media only screen and (min-width: 641px) {
    //tablet
    flex-direction: row;
    justify-content: space-between;
  }

  @media screen and (min-width: 1025px) {
    //desktop
    flex-direction: column;
    justify-content: center;
  }
}

.submit-block {
  text-align: center;
  padding: 2em 0;
  .btn-submit {
    margin: 0 auto;
  }
}
