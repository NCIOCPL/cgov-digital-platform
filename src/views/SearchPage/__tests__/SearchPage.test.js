import React from 'react';
import { render, shallow, mount } from 'enzyme';
import {
  resetDom,
  setupDom
} from '../../../utilities/testHelpers';
import SearchPage from '../SearchPage';

const defaultProps = {
  form: 'basic'
}

const setup = (enzymeMethod = render, props = {}) => {
  const component = enzymeMethod(
    <SearchPage {...props} />
  );
  return component;
};

describe('SearchPage', () => {
  describe('Render', () => {
    let component = null;

    afterEach(() => {
      resetDom();
    });

    it('renders without error', () => {
      component = setup(render, defaultProps);
      expect(component).toMatchSnapshot();
    });
  });

});
