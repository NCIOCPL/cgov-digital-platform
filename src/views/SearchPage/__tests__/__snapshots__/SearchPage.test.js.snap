// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage Render renders without error 1`] = `
<div
  class="general-page-body-container"
>
  <div
    class="contentzone"
  >
    <article
      class="search-page"
    >
      <div
        class="search-page__header"
      >
        <p>
          NCI-supported clinical trials are those sponsored or otherwise financially supported by NCI. See our guide, Steps to Find a Clinical Trial, to learn about options for finding trials not included in NCI's collection.
        </p>
        <div
          class="cts-search-tip"
        >
          <div
            class="cts-search-tip__icon"
          >
            <i
              class="lightbulb-circle-yellow"
            />
          </div>
          <div
            class="cts-search-tip__body"
          >
            <strong>
              Search Tip:
            </strong>
             For more search options, use our 
            <button
              type="button"
            >
              advanced search
            </button>
            .
          </div>
        </div>
      </div>
      <div
        class="search-page__content"
      >
        <form
          class="search-page__form advanced"
        >
          <fieldset
            class="cts-fieldset "
            id="fieldset--type"
          >
            <legend
              class="cts-fieldset__legend"
            >
              <span>
                Cancer Type/Keyword
              </span>
            </legend>
            <a
              aria-label="Help"
              class="text-icon-help"
              href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#basicsearch"
              rel="noopener noreferrer"
              target="_blank"
            >
              ?
            </a>
            <div
              class="cts-fieldset__body"
            >
              <div
                class=""
              >
                <input
                  aria-label="Cancer Type/Keyword"
                  aria-required="false"
                  class="cts-input null"
                  id="q"
                  placeholder="Start typing to select a cancer type"
                  spellcheck="false"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </fieldset>
          <div
            class="side-by-side"
          >
            <fieldset
              class="cts-fieldset "
              id="fieldset--age"
            >
              <legend
                class="cts-fieldset__legend"
              >
                <span>
                  Age
                </span>
              </legend>
              <a
                aria-label="Help"
                class="text-icon-help"
                href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#basicsearch"
                rel="noopener noreferrer"
                target="_blank"
              >
                ?
              </a>
              <div
                class="cts-fieldset__body"
              >
                <div
                  class=""
                >
                  <label
                    class="cts-label"
                    for="a"
                    id="a-label"
                  >
                    Enter the age of the participant.
                  </label>
                  <input
                    aria-labelledby="a-label"
                    aria-required="false"
                    class="cts-input null"
                    id="a"
                    maxlength="3"
                    spellcheck="false"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </fieldset>
            <fieldset
              class="cts-fieldset "
              id="fieldset--zip"
            >
              <legend
                class="cts-fieldset__legend"
              >
                <span>
                  U.S. Zip Code
                </span>
              </legend>
              <a
                aria-label="Help"
                class="text-icon-help"
                href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#basicsearch"
                rel="noopener noreferrer"
                target="_blank"
              >
                ?
              </a>
              <div
                class="cts-fieldset__body"
              >
                <div
                  class=""
                >
                  <label
                    class="cts-label"
                    for="z"
                    id="z-label"
                  />
                  <input
                    aria-labelledby="z-label"
                    aria-required="false"
                    class="cts-input null"
                    id="z"
                    maxlength="5"
                    spellcheck="false"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </fieldset>
          </div>
          <div
            class="submit-block"
          >
            <button
              class="btn-submit"
              type="submit"
            >
              Find Trials
            </button>
          </div>
        </form>
        <aside
          class="search-page__aside"
        >
          <div
            class="cts-delighter-container"
          >
            <div
              class="delighter cts-livehelp"
            >
              <a
                href="/contact"
              >
                <h4>
                  Have a question?
                  <br />
                  We're here to help
                </h4>
                <p>
                  <strong>
                    Chat with us:
                  </strong>
                   LiveHelp
                  <br />
                  <strong>
                    Call us:
                  </strong>
                   1-800-4-CANCER
                  <br />
                  (1-800-422-6237)
                </p>
              </a>
            </div>
            <div
              class="delighter cts-what"
            >
              <a
                href="/about-cancer/treatment/clinical-trials/what-are-trials"
              >
                <h4>
                  What Are Cancer Clinical Trials?
                </h4>
                <p>
                  Learn what they are and what you should know about them.
                </p>
              </a>
            </div>
            <div
              class="delighter cts-which"
            >
              <a
                href="/about-cancer/treatment/clinical-trials/search/trial-guide"
              >
                <h4>
                  Which trials are right for you?
                </h4>
                <p>
                  Use the checklist in our guide to gather the information you’ll need.
                </p>
              </a>
            </div>
          </div>
        </aside>
      </div>
      <div
        class="search-page__footer"
      >
        <div
          class="api-reference-section"
        >
          <h3
            id="ui-id-4"
          >
            The Clinical Trials API: Use our data to power your own clinical trial search
          </h3>
          <p
            class="api-reference-content"
          >
            An application programming interface (API) helps translate large amounts of data in meaningful ways. NCI’s clinical trials search data is now powered by an API, allowing programmers to build applications 
            <a
              href="/syndication/api"
            >
              using this open data.
            </a>
          </p>
        </div>
      </div>
    </article>
  </div>
</div>
`;
